src/Microsoft.AspNet.Http/Features/QueryFeature.cs

```diff
@@ -9,7 +9,7 @@ namespace Microsoft.AspNet.Http.Features.Internal
 {
     public class QueryFeature : IQueryFeature, IFeatureCache
     {
-        private readonly IFeatureCollection _features;
+        private IFeatureCollection _features;
```

Comment:
The changes to these Feature classes can be reverted, especially _features. ResetFeatures doesn't matter much one way or the other.